File: jscomp/test/undef_regression_test.ml


external size_of_t : Obj.t -> 'a Js.undefined = "length" [@@bs.get] 

let f obj = 
  if Js.typeof obj = "function" then 
    ()
  else 
    let size = size_of_t obj in
    match Js.Undefined.toOption size with
    | None -> ()
    | Some s -> Js.log s (* TODO: This case should be peepwholed ..*)

File: jscomp/test/undef_regression_test.js
'use strict';


function f(obj) {
  if (typeof obj === "function") {
    return ;
  }
  var size = obj.length;
  if (size !== void 0) {
    console.log(size);
    return ;
  }
  
}

exports.f = f;
/* No side effect */
